### PR TITLE
Moved the logger initialization into its own implementation file.

### PR DESCRIPTION
**Pull request by dacarley**
Cleaned up plog includes.
Some of the changes are as a result of running clang-format

### CODE REVIEW

**Comment by topisani** [comment]
I don't feel like logger.hpp belongs in util. Otherwise it looks good. Not sure exactly where else it should go though?

**Comment by dacarley** [comment]
The only thing I can think of at the moment would be if we added a "services" folder, sibling to "engines" and "util".  "services" would include logger, state, userconfig...  and other things?  "util" would remain for more generic bits and pieces.

File: src/engines/studio/tapedeck/tapebuffer.cpp [hunk 1]
```diff
@@ -54,6 +54,7 @@ namespace otto {
 
     void main_routine()
     {
+      util::logger::set_thread_name("Tape Buffer");
```
Comment (topisani) [inline src/engines/studio/tapedeck/tapebuffer.cpp]: can we use this thread name for timers too? just an idea

**Comment by topisani** [comment]
loguru looks awesome! I'm merging this, but lets keep thinking about some things here:
1. Where does `logger.[hc]pp` go?
2. the `LOG_F(INFO, ...)` syntax is a little verbose, how about adding `D?LOG[IEF](...)` shorthands?

**Pull request merged**